ir/src/constraints/list_comprehension.rs

```diff
@@ -0,0 +1,342 @@
+use std::collections::BTreeMap;
+
+use super::{graph::CURRENT_ROW, IdentifierType, SemanticError, SymbolTable};
+use air_script_core::{
+    Expression, Identifier, Iterable, ListComprehension, ListFoldingType, NamedTraceAccess,
+    VariableType, VectorAccess,
+};
+
+type IterableContext = BTreeMap<Identifier, Iterable>;
+
+/// Unfolds a list comprehension into a vector of expressions.
+///
+/// Returns an error if there is an error while parsing any of the expressions in the expanded
+/// vector from the list comprehension.
+pub fn unfold_lc(
+    lc: &ListComprehension,
+    symbol_table: &SymbolTable,
+) -> Result<Vec<Expression>, SemanticError> {
+    let num_iterations = get_num_iterations(lc, symbol_table)?;
```

Comment:
Thanks. I've gone with enforcing number of iterations greater than 1 but maybe even 1 should be syntactically correct even if it's not very meaningful so I'm not sure what the right approach is here.

Reply:
In cases like this, I generally prefer the syntactically correct option, because it's more flexible for the user. I could imagine a case where we have a list comprehension as part of a evaluator which is reused over a variety of sets of imported trace column groups, most of which consist of multiple columns, but one of which consists of 1 column. In this case, things would break for the programmer and mean that they need to define a separate function for that one case, whereas it's something that we could probably handle cleanly (especially once there are more optimizations in the IR), allowing them to have a clearer code pattern